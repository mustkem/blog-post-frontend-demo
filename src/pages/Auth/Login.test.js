import React from 'react';
import { shallow } from 'enzyme';
import toJSON  from 'enzyme-to-json';

// import renderer from 'react-test-renderer'

import Login from './Login';

describe("MyComponent", () => {
  it("should render my component", () => {
    shallow(<Login />);
  });
});


//React test renderes. This will render html. 
//   it('should render login component correctly.', () => {  
//     const LoginComponent = renderer.create(<Login />).toJSON();
//     expect(LoginComponent).toMatchSnapshot();
// });

//ths will render coponent tree for this specific component only
it("should render correctly", () => {
    const wrapper = shallow(<Login />)
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  //TODO - Stuff to do 
  /*
  Test handleSubmit.Â 
  Test onChange
  Test input values after entering email and password
  Test Labels
  Test login button if it is rendering
  Test local state, props 
  Mocking api calls
  */