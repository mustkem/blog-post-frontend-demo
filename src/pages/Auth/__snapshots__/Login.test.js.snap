// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<auth>
  <form
    onSubmit={[Function]}
  >
    <input
      control="input"
      id="email"
      label="Your E-Mail"
      onBlur={[Function]}
      onChange={[Function]}
      touched={false}
      type="email"
      valid={false}
      value=""
    />
    <input
      control="input"
      id="password"
      label="Password"
      onBlur={[Function]}
      onChange={[Function]}
      touched={false}
      type="password"
      valid={false}
      value=""
    />
    <button
      design="raised"
      type="submit"
    >
      Login
    </button>
  </form>
</auth>
`;
